src/lib/expansion/expansion-panel-header.ts

```diff
@@ -160,8 +160,8 @@ export class MatExpansionPanelHeader implements OnDestroy, FocusableOption {
    * @param origin Origin of the action that triggered the focus.
    * @docs-private
    */
-  focus(origin: FocusOrigin = 'program') {
-    this._focusMonitor.focusVia(this._element, origin);
+  focus(origin: FocusOrigin = 'program', options?: FocusOptions) {
```

Comment:
Should we make something like:

```ts
interface MatFocusOptions extends FocusOptions {
  origin: FocusOrigin?;
}
```

and then:

```ts
/** @breaking-change 9.0.0 remove FocusOrigin as a param type. */
focus(options: FocusOptions | FocusOrigin = {origin: 'program'}) {...}
```

Reply:
@mmalerba I'm giving this a shot in the step header, it looks like it'll add a fair bit of a code if we want to support it in a backwards-compatible way. In order support the user passing in a `FocusOrigin`, we'd also have to add an optional dependency on the `FocusMonitor`. E.g. here's what the step header would look like:


```ts
focus(config: FocusableOptionConfig = {origin: 'program'}) {
  let origin: FocusOrigin | undefined;
  let options: FocusOptions | undefined;

  if (config == null || typeof config === 'string') {
    origin = config;
  } else {
    origin = config.origin;
    options = config;
  }

  if (this._focusMonitor) {
    this._focusMonitor.focusVia(this._elementRef, origin, options);
  } else {
    this._elementRef.nativeElement.focus(options);
  }
}
```

Reply:
@jelbourn do you have any thoughts? The extra code isn't great, but I also don't like how we wind up with inconsistent APIs otherwise

Reply:
I don't quite follow. Why would you need a dependency on `FocusMonitor` in places where it isn't used now?

Also, FWIW, I've been considering dropping `FocusMonitor` entirely and always showing focus state all the time (recommended by the a11y team and what MDC Web does).

Reply:
@jelbourn it's mostly for consistency.

Reply:
I don't think it's right to add the `FocusMonitor` dependency just for that consistency. I'd rather have the discussion on whether we want to drop `FocusMonitor` now and then maybe just switch everything to the standard options for v8.

Reply:
Ok, then this looks good for now and we can deprecate the origin param later if needed